### PR TITLE
Put current version of jsdom in the root

### PR DESCRIPTION
**Pull request by sandersn**
Previously, the code that supports the oldest version of Typescript was in the root: 3.1 and below. Newer versions were in ts*/ subdirectories.

This PR puts the newest version in the root, and older versions in ts*/ subdirectories. This is possible because all supported versions of the Typescript now understand the typesVersions property in package.json.

This PR needs a new version of DefinitelyTyped-tools and dtslint, which I will ship soon. It also needs to be brought up to date with master.

### CODE REVIEW

File: types/jsdom/jsdom-tests.ts [hunk 1]
```diff
@@ -1,204 +1,18 @@
...
-                    return null;
-                }
-            }
-
-            return super.fetch(url, options);
-        }
-    }
-    new JSDOM('', { resources: new CustomResourceLoader() });
-}
+import '../ts3.5/jsdom-tests';
```
Comment (ExE-Boss) [inline types/jsdom/jsdom-tests.ts:1]:
```suggestion
import './ts3.5/jsdom-tests';
```

File: types/jsdom/ts3.3/jsdom-tests.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import '../jsdom-tests';
+import jsdom = require('jsdom');
+
+const dom = new jsdom.JSDOM();
+const domWindow = dom.window; // $ExpectType DOMWindow
+
+domWindow.document.querySelector('slot'); // $ExpectType HTMLSlotElement | null
+domWindow.AbstractRange.prototype; // $ExpectType AbstractRange
+domWindow.StaticRange.prototype; // $ExpectType StaticRange
+domWindow.CustomElementRegistry.prototype; // $ExpectType CustomElementRegistry
```
Comment (ExE-Boss) [inline types/jsdom/ts3.3/jsdom-tests.ts:10]: This file should have the tests that were in `types/jsdom/jsdom‑tests.ts`.

**Review by ExE-Boss — COMMENTED** [review]
It seems to me like

File: types/jsdom/ts3.3/jsdom-tests.ts (hunk 2 above)
Comment (sandersn) [reply types/jsdom/ts3.3/jsdom-tests.ts]: ts3.3 tests apply to 3.3 - 3.1, so should be moved from ts3.1. I missed deleting `ts3.1` initially, which made this NOT the case. It's fixed now.

File: types/jsdom/jsdom-tests.ts [hunk 3]
```diff
@@ -1,204 +1,18 @@
...
-                    console.log("Ignoring resource requested by iframe element");
-                    return null;
-                }
-            }
-
-            return super.fetch(url, options);
-        }
-    }
-    new JSDOM('', { resources: new CustomResourceLoader() });
-}
```
Comment (ExE-Boss) [inline types/jsdom/jsdom-tests.ts:204]: These test cases shouldn’t have been deleted.

**Pull request merged**